src/include/migraphx/op/if_op.hpp

```diff
@@ -36,6 +36,31 @@ struct if_op
 
         return out_shapes0.front();
     }
+
+    argument compute(
+        const std::vector<argument>& args,
+        const std::vector<module_ref>& mods,
+        const std::function<std::vector<argument>(
+            module_ref& mdl, const std::unordered_map<std::string, argument>& inputs)>& run) const
+    {
+        auto cond      = args.front().at<bool>();
+        module_ref mod = cond ? mods[0] : mods[1];
+        std::unordered_map<std::string, argument> params;
+        // const auto& out_shapes = mod->get_output_shapes();
+        assert(args.size() == 1 or args.size() == mod->get_output_shapes().size() + 1);
+        for(std::size_t i = 1; i < args.size(); ++i)
+        {
+            std::string name = mod->name() + ":#output_" + std::to_string(i - 1);
```

Comment:
The arguments should go to all parameters in the submodule not just the ones named `#output`. It should be the union of all parameters in each module:

```cpp
std::set<std::string> pnames;
for(auto mod:mods)
{
    auto names = mod->get_parameter_names();
    pnames.insert(names.begin(), names.end());
}
assert(pname.size() < args.size());
std::unordered_map<std::string, argument> params;
std::transform(args.begin()+1, 
               args.end(),
               pnames.begin(),
               std::inserter(params, params.end()), 
               [](auto&& arg, auto&& name) {
                   return std::make_pair(name, arg);
               });
```